### DISCUSSION TITLE
How to apply where condition from client side.

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2849] [Q&A] [answered]
I want to apply where directly from client side so no need to hardcode specific fields and condition server side. 

// Model defined

```
const User = objectType({
    name: 'User',
    definition(t) {
        t.model.user_id();
        t.model.user_first_name();
        t.model.user_last_name();
        t.model.user_email();
	},
});
```

I would like to use power of prisma. 
How to sumbit enitre where query directly from client.

Solution 1):

 t.crud.users({
    filtering: true,
    ordering: true,
    pagination: true,
});

```
Client Query:
	query{
		users(where:{user_first_name:{equals:"abc"}}){
	user_id
	}
	}
```

So here I can access  anything using where condition. But unfortunatly its not custom resolver. 

Solution 2):

```
const getAllUser = queryField('getAllUser', {
    type: 'User',
    list: true,
    args: { user_id: intArgs() },
    resolve: async (parent, args, ctx) => {
        try {
            let query = {};
            const users = await ctx.prisma.user.findMany({ where: query });
            return users;
        } catch (error) {
            console.error(error.stack);
            throw error;
        }
    },
});
```
how i can use where condition from client side in custome resolver (solution 2) ??.
I think solution 1 is real power of prisma. but how we can achieve same in custome resolver. ?
Is it possible to do in prisma1?.
Becuase in solution 2 we need to  hardcode different scenario, 
example:
1) string match to first name but not matching to last name 
2) string match to first name and  last name
3)String match to email only.

there can be any number of case. so how we can handle such scenario

**Comment by ryands17** [comment]
Hey @aakashkag :wave: 
You can use a custom resolver in Nexus for crud as well as described [here](https://www.nexusjs.org/#/plugins/prisma?id=resolve). Let me know if this works.

**Comment by aakashkag** [comment]
Hi @ryands17

So i added cutsom logic to test. 
after modification here is solution 1 code.

But custom logic not getting called. it's not printing neither  'logic before the resolver' nor  "logic after the resolver".
Although t.crud returning data as expeted. but cutsom logic not printing anything.
```ts
t.crud.users({
            filtering: true,
            ordering: true,
            pagination: true,
            resolve: async (parent, args, ctx, info, originalResolve) => {
                try {
                    console.log( 'logic before the resolver')
                    const res = await originalResolve(root, args, ctx, info);
                    console.log('logic after the resolver');
                    return res;
                } catch (error) {
                    console.error(error.stack);
                    return error;
                }
            },
        });
```
My prisma client instance sentance is this:
const prisma = new PrismaClient();

document say we require plugin setting("t.crud is an experimental feature. You must explicitly enable it via the plugin options.
").
but without setting also  t.curd is working fine for me.

**Reply by ryands17** [reply]
This is my schema:
```ts
 t.crud.users({
      filtering: true,
      ordering: true,
      resolve: async (root, args, ctx, info, originalResolve) => {
        console.log('logic before the resolver')
        const res = await originalResolve(root, args, ctx, info)
        console.log('logic after the resolver')
        return res
      },
    })
```

and this is the `nexusPrisma` declaration:

```ts
const nexusPrisma = nexusPrismaPlugin({
  experimentalCRUD: true,
  prismaClient: (ctx: Context) => ctx.prisma,
})
```

The above one is working for me, Could you check and see if this works? If possible also could you update `@nexus/schema` and `nexus-prisma` to the latest versions and check.

**Comment by aakashkag** [comment]
```ts
makeSchema is like this.

const schema = makeSchema({
    types: [Query, Mutation, Models],
    plugins: [nexusPrismaPlugin()],
    outputs: {
        schema: join(__dirname, '/generated/schema.graphql'),
        typegen: join(__dirname, '/generated/nexus.ts'),
    },
    typegenAutoConfig: {
        contextType: 'Context.Context',
        sources: [
            {
                source: '.prisma/client',
                alias: 'prisma',
            },
            {
                source: require.resolve('./context'),
                alias: 'Context',
            },
        ],
    },
});

```

**Reply by ryands17** [reply]
Please try the above solution and let me know.

**Reply by aakashkag** [reply]
@ryands17 it's not working for me. 
i created repo here. 
https://github.com/aakashkag/prisma-example
added db dump "./prisma/school-demo.sql"

I am using  "@prisma/cli": "^2.1.1" which is latest i think.

what is it?
  prismaClient: (ctx: Context) => ctx.prisma.
when i am trying to add this line thne i am getting error.

 prisma: (ctx: Context) => ctx.prisma,
^SyntaxError: Unexpected token ':'

if i remove " prisma: (ctx: Context) => ctx.prisma," from constructor then its giving me result. but not displaying on console log any thing. 
here is code logic
https://github.com/aakashkag/prisma-example/blob/master/src/resolvers/all_query_type.js

**Comment by aakashkag** [comment, marked as answer]
Sorry it's working. I upgraded @nexus/schema and @nexus-prisma.

**Comment by ludralph** [comment]
Hi @aakashkag,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
